src/Compilers/CSharp/Portable/Parser/DocumentationCommentParser.cs

```diff
@@ -244,7 +244,7 @@ private XmlNodeSyntax ParseXmlElement()
                                 endName = this.WithXmlParseError(endName, XmlParseErrorCode.XML_InvalidWhitespace);
                             }
 
-                            if (!endName.IsMissing && name.ToString() != endName.ToString())
+                            if (!endName.IsMissing && !MatchingXmlNames(name, endName))
```

Comment:
As much of XML parsing concern the name must be exactly in the form "prefix:name"  with "prefix:" part optional. 
Comparing the whole string is the same as comparing  "name" and "prefix". The colon must be there for the prefix to be parsed as prefix, without whitespaces. 

I will try adding a test, but I am not sure the difference in behavior is actually observable.